frame/nomination-pools/src/tests.rs

```diff
@@ -5100,3 +5215,1045 @@ mod reward_counter_precision {
 			});
 	}
 }
+
+mod commission {
+	use super::*;
+
+	#[test]
+	fn set_commission_works() {
+		ExtBuilder::default().build_and_execute(|| {
+			// Set a commission for pool 1.
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(50), 900))
+			));
+
+			let commission_as_percent = BondedPool::<Runtime>::get(1)
+				.unwrap()
+				.commission
+				.current
+				.as_ref()
+				.map(|(x, _)| *x)
+				.unwrap_or(Perbill::zero());
+			assert_eq!(commission_as_percent, Perbill::from_percent(50));
+
+			// update commission only.
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(25), 900))
+			));
+
+			// update payee only.
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(25), 901))
+			));
+
+			// remove the commission for pool 1.
+			assert_ok!(Pools::set_commission(RuntimeOrigin::signed(900), 1, None));
+
+			// test whether supplying a 0% commission along with a payee results in a None `current`
+			// being inserted.
+			//
+			// set an initial commission of 10%
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(10), 900))
+			));
+			// set the commission to 0%
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(0), 900))
+			));
+			// commssion current should now be None, and `throttle_from` the current block.
+			assert_eq!(
+				BondedPool::<Runtime>::get(1).unwrap().commission,
+				Commission { current: None, max: None, change_rate: None, throttle_from: Some(1) }
+			);
+			// set the commission to 0%
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(0), 900))
+			));
+
+			// test for updating payee only when commission = max commission
+			//
+			// set max commission to 10%
+			assert_ok!(Pools::set_commission_max(
+				RuntimeOrigin::signed(900),
+				1,
+				Perbill::from_percent(10)
+			));
+
+			// set commission to 10%
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(10), 900))
+			));
+
+			// update payee only and keep current commission
+			assert_ok!(Pools::set_commission(
+				RuntimeOrigin::signed(900),
+				1,
+				Some((Perbill::from_percent(10), 901))
+			));
+
+			assert_eq!(
```

Comment:
might be better to call this after each test or bundle of related tests.

Reply:
An event for that was left out as UIs could determine there was a commission change just be listening to the max commission changed event (then compare with current commission in the UI). But yes I can see now there is an argument for including it here too.